src/pymatching/matching.py

```diff
@@ -310,6 +313,76 @@ def load_from_networkx(self, graph: nx.Graph) -> None:
+            detector error model). The `fault_ids` attribute was previously named `qubit_id` in an
+            earlier version of PyMatching, and `qubit_id` is still accepted instead of `fault_ids` in order
+            to maintain backward compatibility.
+            Each ``weight`` attribute should be a non-negative float. If
+            every edge is assigned an error_probability between zero and one,
+            then the ``add_noise`` method can be used to simulate noise and
+            flip edges independently in the graph.
+
+        Examples
+        --------
+        >>> import pymatching
+        >>> import retworkx as rx
+        >>> import math
+        >>> g = rx.PyGraph()
+        >>> matching = g.add_nodes_from([{} for _ in range(3)])
+        >>> edge_a =g.add_edge(0, 1, dict(fault_ids=0, weight=math.log((1-0.1)/0.1), error_probability=0.1))
+        >>> edge_b = g.add_edge(1, 2, dict(fault_ids=1, weight=math.log((1-0.15)/0.15), error_probability=0.15))
+        >>> g[0]['is_boundary'] = True
+        >>> g[2]['is_boundary'] = True
+        >>> m = pymatching.Matching(g)
+        >>> m
+        <pymatching.Matching object with 1 detector, 2 boundary nodes, and 2 edges>
+        """
+        if not isinstance(graph, rx.PyGraph):
+            raise TypeError("G must be a retworkx graph")
+        boundary = {i for i in graph.node_indices() if graph[i].get("is_boundary", False)}
+        num_nodes = len(graph)
+        all_fault_ids = set()
+        g = MatchingGraph(self.num_detectors, boundary)
+        for (u, v, attr) in graph.weighted_edge_list():
+            u, v = int(u), int(v)
+            if "fault_ids" in attr and "qubit_id" in attr:
+                raise ValueError("Both `fault_ids` and `qubit_id` were provided as edge attributes, however use "
+                                 "of `qubit_id` has been deprecated in favour of `fault_ids`. Please only supply "
+                                 "`fault_ids` as an edge attribute.")
+            if "fault_ids" not in attr and "qubit_id" in attr:
+                fault_ids = attr["qubit_id"]  # Still accept qubit_id as well for now
+            else:
+                fault_ids = attr.get("fault_ids", set())
+            if isinstance(fault_ids, (int, np.integer)):
+                fault_ids = {int(fault_ids)} if fault_ids != -1 else set()
+            else:
+                try:
+                    fault_ids = set(fault_ids)
+                    if not all(isinstance(q, (int, np.integer)) for q in fault_ids):
+                        raise ValueError("fault_ids must be a set of ints, not {}".format(fault_ids))
+                except:
+                    raise ValueError(
+                        "fault_ids property must be an int or a set of int"\
+                        " (or convertible to a set), not {}".format(fault_ids))
+            all_fault_ids = all_fault_ids | fault_ids
```

Comment:
There's actually no need to create `all_fault_ids` anymore so this line and line 357 can be removed